compiler.js

```diff
@@ -1,3 +1,299 @@
+let examples = new Map();
+
+examples["helloWorld"] = `// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+contract MyContract {
+    function helloWorld() public pure returns (string memory) {
+        return "Hello, World!";
+    }
+}`;
+
+examples["erc20"] = `// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+contract ERC20 {
+    string public constant name = "ERC20";
+    string public constant symbol = "ERC";
+    uint8 public constant decimals = 18;
+
+    mapping(address => uint256) balances;
+
+    event Transfer(address indexed from, address indexed to, uint256 tokens);
+    
+    uint256 public immutable totalSupply;
+    
+    constructor(uint256 total) {
+        totalSupply = total;
+        balances[msg.sender] = total;
+    }
+
+    function balanceOf(address tokenOwner) public view returns (uint256) {
+        return balances[tokenOwner];
+    }
+    
+    function transfer(address receiver, uint256 numTokens) public returns (bool) {
+        require(balances[msg.sender] >= numTokens);
+        balances[msg.sender] = balances[msg.sender] - numTokens;
+        balances[receiver] = balances[receiver] + numTokens;
+        emit Transfer(msg.sender, receiver, numTokens);
+        return true;
+    }
+    
+    function transferFrom(address owner, address buyer, uint256 numTokens) public returns (bool) {
+        require(balances[owner] >= numTokens);
+        balances[owner] = balances[owner] - numTokens;
+        balances[buyer] = balances[buyer] + numTokens;
+        emit Transfer(owner, buyer, numTokens);
+        return true;
+    }
+}`;
+
+examples["safeRemotePurchase"] = `// SPDX-License-Identifier: GPL-3.0
```

Comment:
This contract has some drawbacks, maybe we should not add it.

Reply:
Removed